### DISCUSSION TITLE
new file is not created with daily_file_sink

### DISCUSSION THREAD

**Discussion by 330305020** [gabime/spdlog#2828] [Q&A] [answered]
I tested the code blow, after 10:07am, a file should be generated, but actually not. What's the reason?

TEST_CASE("daily_logger with dateonly calculator", "[daily_logger]")
{
    using sink_type = spdlog::sinks::daily_file_sink<std::mutex, spdlog::sinks::daily_filename_calculator>;

    prepare_logdir();

    // calculate filename (time based)
    //spdlog::filename_t basename = SPDLOG_FILENAME_T("test_logs/daily_dateonly");
    spdlog::filename_t basename = SPDLOG_FILENAME_T("test_logs/daily_dateonly.log");
    std::tm tm = spdlog::details::os::localtime();
    filename_memory_buf_t w;
    spdlog::fmt_lib::format_to(
        std::back_inserter(w), SPDLOG_FILENAME_T("{}_{:04d}-{:02d}-{:02d}"), basename, tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday);

    auto logger = spdlog::create<sink_type>("logger", basename, 10, 07);
    for (int i = 0; i < 100000; ++i)
    {

        logger->info("Test message {}", i);
        std::this_thread::sleep_for(std::chrono::milliseconds(100)); // new
    }
    logger->flush();

    require_message_count(filename_buf_to_utf8string(w), 10);
}

New message after 10:07am are still writing to the same file.
![image](https://github.com/gabime/spdlog/assets/44339104/a214a5fd-8900-4649-941f-8af843c88051)

**Comment by tt4g** [comment, marked as answer]
daily_file_sink rotates to the "specified time" when the "date changes".
And `spdlog::sinks::daily_filename_calculator` adds only the year, month, and day to the filename, so if it rotates to the same day, nothing will happen because the log will be written to the same file.

https://github.com/gabime/spdlog/blob/2ee8bac78e6525a8ad9a9196e65d502ce390d83a/include/spdlog/sinks/daily_file_sink.h#L26-L39

**Comment by 330305020** [comment]
Thanks for your explanation.
